### PR TITLE
Move Geometry to CSV

### PR DESCRIPTION
**Pull request by GNiendorf**
First commit is calculating the four corners of each sensor directly from the CSV file. I'll push more commits sometime soon.

### CODE REVIEW

**Comment by GNiendorf** [comment]
Still dealing with an issue where the compute_connections does not run correctly with this geom.txt output. I think it's because it expects the coordinates of the four corners in a specific order, but I have to look into it more. Also, there's a step not mentioned in the README which is to run compute_average_layer_geom.py to generate the average_z and average_r txt files. I have to look more into how those are used, but I think the best approach there is to just grab these values directly from the site, i.e. here: https://cms-tklayout.web.cern.ch/cms-tklayout/layouts-work/recent-layouts/OT800_IT615/layout.html

**Comment by GNiendorf** [comment]
https://github.com/SegmentLinking/LSTGeometry/blob/ba49fb56e994f13e13806a17a7bddefd22449e9b/compute_geometry_csv.py#L108-L115

I reverted to using the same modules sizes in the hit-based method instead of the sizes specified by the CSV file. See comparison plots below, the CSV sizes (smaller) increase the fake rate and lower efficiency between abs(eta) 1 to 2.

TC_base_0_0_eff_pt

TC_fakerate_etacoarse

TC_duplrate_etacoarse

**Comment by GNiendorf** [comment]
I'll add a commit that updates the README and post more comparison plots soon.

**Comment by GNiendorf** [comment]
Okay I think this PR is ready for review now. Fixed a bug in the last commit for the old hit based methods caused from the restructuring.

**Comment by GNiendorf** [comment]
Added a simple function to merge the straight/curved module maps after they are computed, rather than having it in a separate file.

**Comment by GNiendorf** [comment]
Pinging @VourMa as a friendly reminder to look at this when you get a chance, thank you!

**Review by VourMa — COMMENTED** [review]
It generally look good, but I must admit that I didn't follow every calculation to the detail in the new files. In any case, my comments are fairly minor.

File: compute_modulemap.py [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+    ref_subdet = Module(ref_detid).subdet()
+    for bound in bounds:
+        helix_p10 = sdlmath.construct_helix_from_points(ptthresh, 0, 0,  10, bound[1], bound[2], bound[0], -charge)
+        helix_m10 = sdlmath.construct_helix_from_points(ptthresh, 0, 0, -10, bound[1], bound[2], bound[0], -charge)
+        helix_p10_pos = sdlmath.construct_helix_from_points(ptthresh, 0, 0,  10, bound[1], bound[2], bound[0], charge)
+        helix_m10_pos = sdlmath.construct_helix_from_points(ptthresh, 0, 0, -10, bound[1], bound[2], bound[0], charge)
+        # helix_p10 = sdlmath.construct_helix_from_points(1, 0, 0,   0, bound[1], bound[2], bound[0], -charge)
+        # helix_m10 = sdlmath.construct_helix_from_points(1, 0, 0,   0, bound[1], bound[2], bound[0], -charge)
+        # helix_p10_pos = sdlmath.construct_helix_from_points(1, 0, 0,   0, bound[1], bound[2], bound[0], charge)
+        # helix_m10_pos = sdlmath.construct_helix_from_points(1, 0, 0,   0, bound[1], bound[2], bound[0], charge)
```
[the remark this answers is not in the document]
Comment (GNiendorf) [reply compute_modulemap.py:132]: These were already commented in the current file, so just left them as commented.

File: merge_module_map.py
[the remark this answers is not in the document]
Comment (GNiendorf) [reply merge_module_map.py:1]: Yes I moved the merge file to just a simple function that gets called in compute_modulemap.py after both are computed.

File: visualization_scripts/visualize_extra.py
[the remark this answers is not in the document]
Comment (GNiendorf) [reply visualization_scripts/visualize_extra.py:1]: Yes these scripts were in the compute_modulemap.py file, but they don't do anything and were just cluttering up the file so I moved them here.

File: README.md
[the remark this answers is not in the document]
Comment (GNiendorf) [reply README.md:1]: Could you explain what you mean by this? Not sure what you're referencing here.

[the remark this answers is not in the document]
Comment (VourMa) [reply README.md:1]: image

In these blocks, I thought that you meant to have all text in normal format and only the phrases between ` in code format.

[the remark this answers is not in the document]
Comment (GNiendorf) [reply README.md:1]: Thanks, fixed.

**Review by VourMa — APPROVED** [review]
Everything look good, thank you @GNiendorf!

**Pull request merged**